src/messaging/messaging.ts

```diff
@@ -332,6 +339,40 @@ function validateAps(aps: Aps) {
   }
 }
 
+function validateApsSound(sound: string | CriticalSound) {
+  if (typeof sound === 'undefined' || validator.isString(sound)) {
+    return;
+  } else if (!validator.isNonNullObject(sound)) {
+    throw new FirebaseMessagingError(
+      MessagingClientErrorCode.INVALID_PAYLOAD,
+      'apns.payload.aps.sound must be a string or a non-null object');
+  }
+
+  const volume = (sound as CriticalSound).volume;
```

Comment:
Does typescript not narrow it down to CriticalSound at this point? 
If it's the case, you can define `validator.isString` as an user defined type guard with signature:
`isString(input: any): input is String`

Reply:
Done